src/devices/Bmxx80/Bme280.cs

```diff
@@ -73,12 +74,15 @@ public void SetHumiditySampling(Sampling sampling)
         /// <returns>Returns a percentage from 0 to 100.</returns>
         public async Task<double> ReadHumidityAsync()
         {
+            if (HumiditySampling == Sampling.Skipped)
```

Comment:
When you provide async method you should as well provide synchronous version of the same function

Reply:
I would actually change these to synchronous methods and not provide asynchronous ones thinking about it. These methods just have an async signature and no real async behavior. The signature was chosen to keep in line with the Bmx280 implementation.

But the Bme680 actually works a little different compared to the Bmx280 right now:

```C#
...
var measurementDuration = bme680.GetMeasurementDuration(bme680.HeaterProfile);
bme680.SetPowerMode(PowerMode.Forced);          // kick off measurement
Thread.Sleep(measurementDuration);              // wait for measurement
var temp = await bme680.ReadTemperatureAsync(); // this just reads from the register
...
```

compared to:

```C#
...
bmx280.SetPowerMode(Bmx280PowerMode.Forced);          // kick off measurement
var tempValue = await bmx280.ReadTemperatureAsync();  // wait and read after
...
```
The behavior of the bmx280 is actually a little bit suboptimal since the ReadHumidityAsnyc (same for Pressure) works like this:

```C#
if (ReadPowerMode() == Bmx280PowerMode.Forced)
{
    await Task.Delay(GetMeasurementTimeForForcedMode(HumiditySampling));
}

await ReadTemperatureAsync();

byte msb = Read8BitsFromRegister((byte)Bme280Register.HUMIDDATA_MSB);
byte lsb = Read8BitsFromRegister((byte)Bme280Register.HUMIDDATA_LSB);
```

So it always queries the power mode first to check if a measurement is still being taken and then waits for the time the current measurement will take. This breaks down for instance if:

TemperatureSampling = X1, PressureSampling = X1, HumiditySampling = X16

If you call

```C#
...
bmx280.SetPowerMode(Bmx280PowerMode.Forced);          // this will take 7 + 7 + 44ms
var tempValue = await bmx280.ReadTemperatureAsync();  // this will only wait for 7ms
...
```

So this actually doesn't make sense, I just did not think about this until further looking into why this method signature is actually async.
I would change the behavior of the Bmx280 to the one of the Bme680 if you agree and drop the async signatures.

Reply:
Yes, I agree async is not particularly useful here